### PR TITLE
chia|tests|pre-commit: Add missing `__init__.py` files

### PR DESCRIPTION
**Pull request by xdustinface**

### CODE REVIEW

**Comment by xdustinface** [comment]
I'm not a python expert either. I just run into some situation where PyCharm didn't find files in folders without the `__init__.py` when i used the search "In Project" vs "Directory" if i recall correct and it also highlights them different, see 
Screenshot from 2021-10-07 17-44-01

Any concerns about it?

**Comment by richardkiss** [comment]
I'm not an expert on best practices for 3.x. I know that Python 2 required `__init__.py` files and Python 3 does not, but I don't have a completely clear idea why or when we require them. My intuitive preference would be to not add them if they're not needed (because they're not needed), but I'm also very willing to believe that they affect `import`.

**Review by jack60612 — DISMISSED** [review]
In my opinion the init file addition causes no harm and can help alleviate problems when using an ide so it should be fine.

**Review by altendky — CHANGES_REQUESTED** [review]
Just chatted with Richard.  Let's make it roughly all `chia/**` directories and `tests/**` directories but not in `benchmarks/`.  That leaves the following to be changed.

- Remove `benchmarks/__init__.py`
- Add `chia/wallet/puzzles/prefarm/__init__.py`

**Review by altendky — DISMISSED** [review]

**Comment by altendky** [comment]
I think this needs a pre-commit hook.  There's at least a first pass at that in https://github.com/xdustinface/chia-blockchain/pull/1.

File: tests/build-init-files.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            logger.warning(f"Created : {init_path}")
+        else:
+            failed = True
+            logger.error(f"Fail    : present but not a regular file: {init_path}", file=sys.stderr)
+
+    if failed:
+        raise click.ClickException("At least one __init__.py created or not a regular file")
+
+
+command()
```
Comment (altendky) [inline tests/build-init-files.py:56]:
```
************* Module tests.build-init-files
tests/build-init-files.py:56:0: E1120: No value for argument 'verbose' in function call (no-value-for-parameter)
tests/build-init-files.py:56:0: E1120: No value for argument 'root_str' in function call (no-value-for-parameter)
```

```suggestion
command()  # noqa: E1120
```

File: tests/build-init-files.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            logger.warning(f"Created : {init_path}")
+        else:
+            failed = True
+            logger.error(f"Fail    : present but not a regular file: {init_path}", file=sys.stderr)
+
+    if failed:
+        raise click.ClickException("At least one __init__.py created or not a regular file")
+
+
+command()  # noqa: E1120
```
Comment (altendky) [inline tests/build-init-files.py:56]: Sorry about that, I added a flake8 ignore last time.  This seems to work locally as a pylint ignore...

```suggestion
command()  # pylint: disable=no-value-for-parameter
```

File: chia/plotting/create_plots.py [hunk 3]
```diff
@@ -42,10 +42,10 @@ def pool_contract_puzzle_hash(self) -> Optional[bytes32]:
...
     def __init__(
         self,
-        farmer_public_key: str,
-        alt_fingerprint: int,
-        pool_public_key: str,
-        pool_contract_address: str,
+        farmer_public_key: Optional[str],
+        alt_fingerprint: Optional[int],
+        pool_public_key: Optional[str],
+        pool_contract_address: Optional[str],
```
Comment (richardkiss) [inline chia/plotting/create_plots.py:48]: Why is this change included? I thought this was about adding `__init__.py` files?

File: tests/plotting/test_plot_manager.py [hunk 4]
```diff
@@ -488,7 +488,7 @@ def raising_callback(event: PlotRefreshEvents, _: PlotRefreshResult):
     assert len(env.refresh_tester.plot_manager.no_key_filenames) == 0
     # The next run without the valid callback should lead to re-loading of all plot
     env.refresh_tester.plot_manager.set_refresh_callback(default_callback)
-    expected_result.loaded = env.dir_1.plot_info_list() + env.dir_2.plot_info_list()
+    expected_result.loaded = env.dir_1.plot_info_list() + env.dir_2.plot_info_list()  # type: ignore[assignment]
```
Comment (richardkiss) [inline tests/plotting/test_plot_manager.py:491]: Ditto... why is this here?

File: chia/plotting/create_plots.py (hunk 3 above)
Comment (xdustinface) [reply chia/plotting/create_plots.py:48]: Adding `chia/plotting/__init__.py` enabled more `mypy` checks in the related code.

File: tests/plotting/test_plot_manager.py (hunk 4 above)
Comment (xdustinface) [reply tests/plotting/test_plot_manager.py:491]: Adding `chia/plotting/__init__.py` enabled more `mypy` checks in the related code.

**Review by richardkiss — APPROVED** [review]
